### PR TITLE
Replace all std with alloc except for std::error::Error

### LINKED ISSUE #58: Replace std uses with alloc
Since [Rust 1.36.0](https://blog.rust-lang.org/2019/07/04/Rust-1.36.0.html), `alloc` is usable in `no_std` environments.

Granted, I'm pretty sure the funxions using `alloc` funxionality should still be feature-gated, but probably with `alloc` instead of `std`.

### PR DESCRIPTION
**Pull request by nabijaczleweli**
With only one hiccup!

Closes #58

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -31,11 +31,11 @@
 /// # Ok(())
 /// # }
 /// ```
-#[cfg(feature = "std")]
+#[cfg(feature = "alloc")]
 #[macro_export]
 macro_rules! try_copy {
     ($res:expr) => {{
-        use std::borrow::Cow;
+        use $crate::alloc::borrow::Cow;  // TODO: There *has* to be a better way of doing this, right? (also below)
```
Comment (nabijaczleweli) [inline src/util.rs]: *Is* there a better way of doing this?

Comment (hellow554) [reply src/util.rs]: `::alloc::borrow`?

**Comment by nabijaczleweli** [comment]
Ah, yeah, I must've used the wrong arguments when testing previously.

The doctests say [error](https://travis-ci.org/nabijaczleweli/safe-transmute-rs/jobs/555796199#L529) now, too, and *before* any of them get compiled, *but* they still pass, so false negative?

**Comment by nabijaczleweli** [comment]
[Confirmed false negative](https://github.com/rust-lang/rust/issues/52243)

**Comment by nabijaczleweli** [comment]
Released in [v0.11.0-rc.2](https://github.com/nabijaczleweli/safe-transmute-rs/releases/tag/v0.11.0-rc.2)

**Pull request merged**